presto-main/src/test/java/com/facebook/presto/type/TestDecimalParametricType.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.type;
+
+import org.testng.annotations.Test;
+
+public class TestDecimalParametricType
+        extends AbstractTestDecimalFunctions
+{
+    @Test
+    public void decimalIsCreatedWithPrecisionAndScaleDefined()
+    {
+        assertDecimalFunction("CAST(1 AS DECIMAL(2, 0))", decimal("01"));
+        assertDecimalFunction("CAST(0.01 AS DECIMAL(2, 2))", decimal(".01"));
+        assertDecimalFunction("CAST(0.02 AS DECIMAL(10, 10))", decimal(".0200000000"));
+        assertDecimalFunction("CAST(0.02 AS DECIMAL(10, 8))", decimal("00.02000000"));
```

Comment:
Maybe add a test with precision causing truncation?
E.g. CAST(2.3 AS DECIMAL(2,0)) -> expected 2.0
I know there is such a test in tests for decimal without parameters given but there is an implicit assumption of precision and scale so I claim the explicit test here would be something appreciated in addition to the other one.